modules/telestion-application/src/main/java/de/wuespace/telestion/application/launcher/TelestionConfigParser.java

```diff
@@ -0,0 +1,183 @@
+package de.wuespace.telestion.application.launcher;
+
+import de.wuespace.telestion.application.loader.Loader;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Parses a Telestion configuration in the {@link JsonObject} format.
+ * It extracts different configuration options like {@link Loader loaders} or {@link VertxOptions Vert.x options}.
+ * <p>
+ * It's mainly used by the {@link TelestionLauncher} class.
+ *
+ * @author Ludwig Richter
+ */
+@SuppressWarnings("ClassCanBeRecord")
+public class TelestionConfigParser {
+
+	/**
+	 * The name of the JSON property which points the {@link Vertx#isClustered()}  is clustered option}
+	 * in the Telestion configuration.
+	 */
+	public static final String IS_CLUSTERED_KEY = "isClustered";
+
+	/**
+	 * The name of the JSON property which points the {@link VertxOptions Vert.x options}
+	 * in the Telestion configuration.
+	 */
+	public static final String VERTX_OPTIONS_KEY = "vertxOptions";
+
+	/**
+	 * The name of the JSON property which points to {@link Loader loaders} that should be loaded
+	 * by the {@link TelestionLauncher} in the Telestion configuration.
+	 */
+	public static final String LOADERS_KEY = "loaders";
+
+	/**
+	 * The name of the JSON property which points to the classname of the {@link Loader loader}
+	 * in the loaders' entry.
+	 */
+	public static final String EXTENDED_LOADER_CLASSNAME_KEY = "className";
+
+	/**
+	 * The name of the JSON property which points to the {@link Loader loader} configuration
+	 * in the loaders' entry.
+	 */
+	public static final String EXTENDED_LOADER_CONFIG_KEY = "config";
+
+	/**
+	 * The received {@link JsonObject} containing the raw Telestion configuration received upon creation.
```

Comment:
Do you have a javadoc link to Telestion configuration?

Reply:
Sorry, there is currently no Telestion Configuration type we can reference to.